src/nvim/syntax.c

```diff
@@ -6569,8 +6569,8 @@ void do_highlight(const char *line, const bool forceit, const bool init)
       restore_cterm_colors();
 
       // Clear all default highlight groups and load the defaults.
-      for (int idx = 0; idx < highlight_ga.ga_len; idx++) {
-        highlight_clear(idx);
+      for (int idx_ = 0; idx_ < highlight_ga.ga_len; idx_++) {
```

Comment:
why not just `int i` ?  predeclared `long i` can be moved to just before the loop where it is used.

Reply:
`i` is used all over this function and in some cases the result is used later. Then if a vim patch uses `i` we have to again scrutinize the flow.

Reply:
`i` is used locally in two clearly delimited blocks. But call it `j` if you are worried about vim-patches. The point was that there is nothing special about `idx` (or any relation to the outer `idx`) that makes it need be semi-preserved as `idx_`.

Reply:
There's also nothing particularly bad about `idx_` .

Reply:
It is not a big deal, but it does look uglier, especially with postfix operatorlike `idx_++` (looks like a join between `idx` and `++`), and it is harder to pronounce the difference between `x` and `x_`. 

`idx`, `i` and `j` is the most common pattern for multiple "anonymous" indexes, this would deviate from that for no particular reason. It would be better to reserve `x_` for where it actually means something (the existing cases I have seen are replacement for reserved keyword, and barebones marco hygine, of which the later only works if it is not randomly used for other stuff...).